### PR TITLE
[IMP] mail: make the activity popover test more reliable

### PR DESCRIPTION
**Pull request by phenix-factory**
This PR make the activity popover test use the mock model and a chatter as
component.

### CODE REVIEW

File: addons/mail/static/src/components/activity_mark_done_popover/tests/activity_mark_done_popover_tests.js [hunk 1]
```diff
@@ -41,14 +42,26 @@ QUnit.module('activity_mark_done_popover_tests.js', {
...
+        res_model: 'res.partner',
+    });
+    const { createChatterContainerComponent } = await this.start();
+    await createChatterContainerComponent({
+        threadId: 100,
+        threadModel: 'res.partner',
+    });
+
+    await afterNextRender(() => {
+        document.querySelector('.o_Activity_markDoneButton').click();
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity_mark_done_popover/tests/activity_mark_done_popover_tests.js:63]: I've recently added a new `click()` on `start` res to simplify these, those on which you have diff would be nice to use it.

Comment (phenix-factory) [reply addons/mail/static/src/components/activity_mark_done_popover/tests/activity_mark_done_popover_tests.js]: Alright !

File: addons/mail/static/src/components/activity_mark_done_popover/tests/activity_mark_done_popover_tests.js [hunk 2]
```diff
@@ -260,26 +304,33 @@ QUnit.test('[technical] activity mark done & schedule next with new action', asy
...
+                return this._super(...arguments);
+            },
+            env: { bus },
+    });
+    await createChatterContainerComponent({
+        threadId: 100,
+        threadModel: 'res.partner',
     });
+    await click('.o_Activity_markDoneButton');
+
```
Comment (alexkuhn) [inline addons/mail/static/src/components/activity_mark_done_popover/tests/activity_mark_done_popover_tests.js:332]: no newline between 2 successive clicks

**Review by alexkuhn — APPROVED** [review]
LGTM

File: addons/mail/static/src/components/activity_mark_done_popover/tests/activity_mark_done_popover_tests.js [hunk 3]
```diff
@@ -260,26 +303,33 @@ QUnit.test('[technical] activity mark done & schedule next with new action', asy
...
-        thread: insert({ id: 42, model: 'res.partner' }),
+        res_id: 100,
+        res_model: 'res.partner',
     });
-    await this.createActivityMarkDonePopoverComponent(activity);
-
-    await afterNextRender(() => {
-        document.querySelector('.o_ActivityMarkDonePopover_doneScheduleNextButton').click();
+    const { click, createChatterContainerComponent } = await this.start({
+            async mockRPC(route, args) {
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity_mark_done_popover/tests/activity_mark_done_popover_tests.js:318]: wrong indent

File: addons/mail/static/src/components/activity_mark_done_popover/tests/activity_mark_done_popover_tests.js [hunk 4]
```diff
@@ -260,26 +303,33 @@ QUnit.test('[technical] activity mark done & schedule next with new action', asy
...
+                if (route === '/web/dataset/call_kw/mail.activity/action_feedback_schedule_next') {
+                    return { type: 'ir.actions.act_window' };
+                }
+                return this._super(...arguments);
+            },
+            env: { bus },
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity_mark_done_popover/tests/activity_mark_done_popover_tests.js:324]: while we had diff, env should be before mockRPC

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Comment by seb-odoo** [comment]
@robodoo retry

**Pull request closed without merging**